tests/jeos/check_efi_setup.pm

```diff
@@ -0,0 +1,271 @@
+    if (exists($h->{exec}) && defined($h->{exec})) {
+        $h->{exec} =~ s'\\'/'g;
+    } else {
+        $h->{exec} = undef;
+    }
+
+    return $h;
+}
+
+sub check_efi_state {
+    is_efi_boot or die "Image did not boot in UEFI mode!\n";
+    my $expected = shift;
+
+    # check UEFI firmware bitness
+    validate_script_output "cat ${ \SYSFS_EFI_BITS }", sub { $_ == 64 };
+
+    # check SecureBoot according to efivars
+    # get data from efivars
+    # {8be4df61-93ca-11d2-aa0d-00e098032b8c} {global} efi_guid_global EFI Global Variable
+    # save only the first capture
+    my ($efi_guid_global, undef) = script_output('efivar --list-guids') =~ /\{((\w+-){4}\w+)\}.*\s+efi_guid_global\s+/;
+    diag "Found efi guid=$efi_guid_global";
+    diag('Expected state of SecureBoot: ' . get_var('DISABLE_SECUREBOOT', 0) ? 'Disabled' : 'Enabled');
+    validate_script_output
+      "od --address-radix=n --format=u1 /sys/firmware/efi/vars/SecureBoot-$efi_guid_global/data",
+      sub { $_ == !get_var('DISABLE_SECUREBOOT', 0) };
+
+    # get current boot information from efibootmgr
+    my $efi = efibootmgr_current_boot;
+
+    diag("Found:\nlabel=$efi->{label}\nexecutable=$efi->{exec}");
+    diag("Expected:\nlabel=$expected->{label}\nexecutable=$expected->{exec}");
+
+    if (exists $expected->{exec} && defined $expected->{exec}) {
+        record_info "Expected", "EFI executable: $expected->{exec} ( $expected->{label} )";
+        if (exists $efi->{exec} && defined $efi->{exec}) {
+            ($efi->{exec} eq $expected->{exec} && $efi->{label} eq $expected->{label}) or die "System booted using unexpected efi binary\n";
+        }
+        record_info "Found", "EFI executable: $efi->{exec} ( $efi->{label} )";
+    } else {
+        record_info "Fallback", "EFI label: $efi->{label}";
+        return;
+    }
+
+    my $issuer_regex = is_opensuse() ? qr/^\s+issuer:\s+\/CN=openSUSE Secure Boot CA/ :
+      qr/^\s+issuer:\s+\/CN=SUSE Linux Enterprise Secure Boot CA/;
+    diag("Verifying shim's certificate");
+    (grep /$issuer_regex/, split /\n/, script_output "sbverify --list $expected->{mount}/$efi->{exec}") or
+      die "No SUSE/openSUSE certificate has been found in sbverify's output";
+    assert_script_run 'openssl x509 -in $(rpm -ql shim | grep crt) -inform DER -text -out shim.crt';
+    assert_script_run "sbverify --cert shim.crt $expected->{mount}/$efi->{exec}";
+
+    if (script_run "cat /proc/keys | tee -a /dev/$serialdev | grep asym") {
+        if (is_leap('15.1')) {
+            record_soft_failure('boo#1170896 - secureboot keys not found in /proc/keys');
+        } else {
+            die "No asymetric keys in keyring";
+        }
+    }
+
+}
+
+sub check_mok {
+    my $state = shift() ? qr/^SecureBoot\senabled$/ : qr/^SecureBoot\sdisabled$/;
+    # check SecureBoot according to MOK
+    diag('Expected regex used to verify SecureBoot: ' . $state);
+    validate_script_output 'mokutil --sb-state', sub { $state };
```

Comment:
I think this should work without sub:
```suggestion
    validate_script_output 'mokutil --sb-state', $state;
```
see: https://github.com/os-autoinst/os-autoinst/blob/master/testapi.pm#L1208

Reply:
Nice catch, thanks!